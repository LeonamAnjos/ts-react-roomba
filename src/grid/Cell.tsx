import React, { useContext } from "react";
import { RoombaContext } from "../contexts/RoombaContext";
import "../styles/Grid.css";

type RowProps = {
  column: number;
  row: number;
};

const Row = ({ column, row }: RowProps) => {
  const { position, getEmoji } = useContext(RoombaContext);

  const isRoombaPresent = position.column === column && position.row === row;

  return <div className="Cell">{isRoombaPresent && getEmoji()}</div>;
};

export default Row;
