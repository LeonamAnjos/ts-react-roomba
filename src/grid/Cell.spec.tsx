import { render, screen, fireEvent } from "@testing-library/react";
import { ReactNode, useContext } from "react";
import RoombaContextProvider, {
  RoombaContext,
} from "../contexts/RoombaContext";
import Cell from "./Cell";

type TestCellProps = { column: number; row: number };

const TestCell = ({ column, row }: TestCellProps) => {
  const { rotateRight } = useContext(RoombaContext);

  const handleRotateClick = () => rotateRight();

  return (
    <>
      <button data-testid={"direction-button"} onClick={handleRotateClick}>
        Rotate
      </button>
      <Cell column={column} row={row} />
    </>
  );
};

describe("<Cell />", () => {
  const customRender = (children: ReactNode) => {
    return render(
      <RoombaContextProvider size={4}>{children}</RoombaContextProvider>
    );
  };
  it("SHOULD render WHEN cell is populated", () => {
    const { container } = customRender(<TestCell column={0} row={0} />);

    const element = screen.getByTestId("direction-button");

    expect(container).toMatchSnapshot();

    fireEvent.click(element);
    expect(container).toMatchSnapshot();

    fireEvent.click(element);
    expect(container).toMatchSnapshot();

    fireEvent.click(element);
    expect(container).toMatchSnapshot();
  });

  it("SHOULD render WHEN initial state AND cell is empty", () => {
    const { container } = customRender(<TestCell column={1} row={1} />);

    expect(container).toMatchSnapshot();
  });
});
