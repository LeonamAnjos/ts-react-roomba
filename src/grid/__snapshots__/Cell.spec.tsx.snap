// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> SHOULD render WHEN cell is populated 1`] = `
<div>
  <button
    data-testid="direction-button"
  >
    Rotate
  </button>
  <div
    class="Cell"
  >
    ðŸ‘‰
  </div>
</div>
`;

exports[`<Cell /> SHOULD render WHEN cell is populated 2`] = `
<div>
  <button
    data-testid="direction-button"
  >
    Rotate
  </button>
  <div
    class="Cell"
  >
    ðŸ‘‡
  </div>
</div>
`;

exports[`<Cell /> SHOULD render WHEN cell is populated 3`] = `
<div>
  <button
    data-testid="direction-button"
  >
    Rotate
  </button>
  <div
    class="Cell"
  >
    ðŸ‘ˆ
  </div>
</div>
`;

exports[`<Cell /> SHOULD render WHEN cell is populated 4`] = `
<div>
  <button
    data-testid="direction-button"
  >
    Rotate
  </button>
  <div
    class="Cell"
  >
    ðŸ‘†
  </div>
</div>
`;

exports[`<Cell /> SHOULD render WHEN initial state AND cell is empty 1`] = `
<div>
  <button
    data-testid="direction-button"
  >
    Rotate
  </button>
  <div
    class="Cell"
  />
</div>
`;
